### PR TITLE
SOC-6027 : secure REST endpoints to get people infos and to manage connections

### PR DESCRIPTION
**Pull request by thomasdelhomenie**
The REST endpoints to get people infos (/rest/social/people) are publicly available and REST endpoints used to manage connections and space invitations (/rest/social/notifications) are not correctly protected.
This fix adds the protection to these endpoints:
* /rest/social/people endpoints are now accessible to authenticated users only
* /rest/social/notifications endpoints are accessible only to users who can do the operations

### CODE REVIEW

File: component/service/src/main/java/org/exoplatform/social/service/rest/PeopleRestService.java [hunk 1]
```diff
@@ -597,6 +597,7 @@ private void addSpaceOrUserToList(List<Identity> identities, IdentityNameList op
    */
   @GET
   @Path("getprofile/data.json")
+  @RolesAllowed("users")
```
Comment (boubaker) [inline component/service/src/main/java/org/exoplatform/social/service/rest/PeopleRestService.java]: IMO, we should add this annotation to the class level

File: component/service/src/main/java/org/exoplatform/social/service/rest/NotificationsRestService.java [hunk 2]
```diff
@@ -42,9 +42,9 @@
 import javax.ws.rs.core.UriInfo;
 import java.net.URI;
 import java.util.Arrays;
-import java.util.List;
 
 import static org.exoplatform.social.service.rest.RestChecker.checkAuthenticatedRequest;
+import static org.exoplatform.social.service.rest.RestChecker.checkAuthenticatedUserPermission;
```
Comment (boubaker) [inline component/service/src/main/java/org/exoplatform/social/service/rest/NotificationsRestService.java]: IMO we should add RolesAllowed("users") annotation to class level

File: component/service/src/main/java/org/exoplatform/social/service/rest/PeopleRestService.java (hunk 1 above)
Comment (thomasdelhomenie) [reply component/service/src/main/java/org/exoplatform/social/service/rest/PeopleRestService.java]: Done

File: component/service/src/main/java/org/exoplatform/social/service/rest/NotificationsRestService.java (hunk 2 above)
Comment (thomasdelhomenie) [reply component/service/src/main/java/org/exoplatform/social/service/rest/NotificationsRestService.java]: You mean adding RolesAllowed("users") besides the existing checks ?

Comment (boubaker) [reply component/service/src/main/java/org/exoplatform/social/service/rest/NotificationsRestService.java]: Nope, the checks are great, just add an additional check on role for class

Comment (thomasdelhomenie) [reply component/service/src/main/java/org/exoplatform/social/service/rest/NotificationsRestService.java]: ok, done.

**Pull request merged**